virtcontainers/api.go

```diff
@@ -931,3 +931,52 @@ func ListRoutes(ctx context.Context, sandboxID string) ([]*vcTypes.Route, error)
 
 	return s.ListRoutes()
 }
+
+// CleanupContaienr is used by shimv2 to stop and delete a container exclusively, once there is no container
+// in the sandbox left, do stop the sandbox and delete it. Those serial operations will be done exclusively by
+// locking the sandbox.
+func CleanupContainer(ctx context.Context, sandboxID, containerID string) error {
+	span, ctx := trace(ctx, "ListRoutes")
+	defer span.Finish()
+
+	if sandboxID == "" {
+		return vcTypes.ErrNeedSandboxID
+	}
+
+	lockFile, err := rwLockSandbox(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+	defer unlockSandbox(ctx, sandboxID, lockFile)
+
+	s, err := fetchSandbox(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+
+	s.stateful = false
+	s.releaseStatelessSandbox()
```

Comment:
I think I understand this: `releaseStatelessSandbox()` only calls `Release()` for stateless sandboxes. But `releaseStatelessSandbox()` is a "private" sandbox method (since it starts with a lower-case letter), but bizarrely it calls a public method `Release()`. So I'm wondering if you should just call `s.Release()` here? Maybe this is better (safer) in case the implementation of `releaseStatelessSandbox()` ever changes though.

Reply:
Yes, you are right.